pqarrow/parquet.go

```diff
@@ -37,6 +37,11 @@ func ArrowScalarToParquetValue(sc scalar.Scalar) (parquet.Value, error) {
 }
 
 func appendToRow(row []parquet.Value, c arrow.Array, index, rep, def, col int) ([]parquet.Value, error) {
+	if c.IsNull(index) {
+		row = append(row, parquet.ValueOf(nil).Level(rep, def, col))
```

Comment:
Will it be possible to append a value with a def level > 0? I just ran into a subtle bug where we were doing this in logic test value creation. Parquet considers values null if the definition level != max definition level: https://github.com/segmentio/parquet-go/blob/main/column_buffer.go#L297

Reply:
Yes, will fix.